### PR TITLE
Prevent early connection closure over https #1219

### LINKED ISSUE #1219: Large response entity is truncated over https with 'Connection: close' header
When a client sends `Connection: close` header and response entity is large, the entity gets truncated over `https`.  It is ok with `http`.  We see this behaviour only with larger entities.  Please see below reproducer:

```scala
import java.io.InputStream
import java.security.{KeyStore, SecureRandom}
import javax.net.ssl.{KeyManagerFactory, SSLContext, TrustManagerFactory}

import akka.actor.ActorSystem
import akka.http.scaladsl.server.Route
import akka.http.scaladsl.{ConnectionContext, Http, HttpsConnectionContext}
import akka.stream.ActorMaterializer
import akka.http.scaladsl.server.Directives._

import scala.util.Random

object HttpsWithLargeResponseEntity extends App {

  implicit val system = ActorSystem()
  implicit val mat = ActorMaterializer()
  implicit val dispatcher = system.dispatcher

  // Manual HTTPS configuration

  val password: Array[Char] = "changeit".toCharArray // do not store passwords in code, read them from somewhere safe!

  val ks: KeyStore = KeyStore.getInstance("JKS")
  val keystore: InputStream = getClass.getClassLoader.getResourceAsStream("example.com.jks")

  require(keystore != null, "Keystore required!")
  ks.load(keystore, password)

  val keyManagerFactory: KeyManagerFactory = KeyManagerFactory.getInstance("SunX509")
  keyManagerFactory.init(ks, password)

  val tmf: TrustManagerFactory = TrustManagerFactory.getInstance("SunX509")
  tmf.init(ks)

  val sslContext: SSLContext = SSLContext.getInstance("TLS")
  sslContext.init(keyManagerFactory.getKeyManagers, tmf.getTrustManagers, new SecureRandom)
  val https: HttpsConnectionContext = ConnectionContext.https(sslContext)

  val routes: Route = get { complete(Random.nextString(1024 * 100)) }
  Http().bindAndHandle(routes, "127.0.0.1", 8443, connectionContext = https)
  Http().bindAndHandle(routes, "127.0.0.1", 8080)
}
```

With the following commands, I see the entire entity is retrieved:

* `curl -kv https://localhost:8443/`
* `curl -v http://localhost:8080/ -H 'Connection: close'`
* `curl -v http://localhost:8080/`

However, when the `Connection: close'` header is sent and the protocol is `https`, the entity gets truncated:

* `curl -kv https://localhost:8443/ -H 'Connection: close'`

```
transfer closed with 90231 bytes remaining to read
* Curl_http_done: called premature == 1
* Closing connection 0
curl: (18) transfer closed with 90231 bytes remaining to read
```

### PR DESCRIPTION
**Pull request by anilgursel**
Fixes #1219

### CODE REVIEW

**Comment by jlprat** [comment]
OK TO TEST

**Comment by jlprat** [comment]
refs: #1219

File: akka-http-core/src/test/scala/akka/http/scaladsl/ClientServerSpec.scala [hunk 1]
```diff
@@ -536,6 +538,37 @@ class ClientServerSpec extends WordSpec with Matchers with BeforeAndAfterAll wit
       } finally server.foreach(_.unbind())
     }
 
+    "complete a request/response over https when request has `Connection: close` set" in Utils.assertAllStagesStopped {
+      // akka/akka-http#1219
+      val serverToClientNetworkBufferSize = 1000
```
Comment (jrudolph) [inline akka-http-core/src/test/scala/akka/http/scaladsl/ClientServerSpec.scala]: :+1: Good one to trigger a streamed entity.

File: akka-http-core/src/test/scala/akka/http/scaladsl/ClientServerSpec.scala [hunk 2]
```diff
@@ -536,6 +538,37 @@ class ClientServerSpec extends WordSpec with Matchers with BeforeAndAfterAll wit
       } finally server.foreach(_.unbind())
     }
 
+    "complete a request/response over https when request has `Connection: close` set" in Utils.assertAllStagesStopped {
```
Comment (jrudolph) [inline akka-http-core/src/test/scala/akka/http/scaladsl/ClientServerSpec.scala]: I checked that the test fails without the fix.

**Pull request merged**